### PR TITLE
Handle nanosecond precision in TestInstantConversion

### LINKED ISSUE #3945: GC test TestInstantConversion fails on Java 17
```
[ERROR] org.projectnessie.gc.base.TestInstantConversion.testInstantConversion  Time elapsed: 0.173 s  <<< FAILURE!
org.opentest4j.AssertionFailedError: 

expected: 2022-04-13T07:42:09.706331353Z
 but was: 2022-04-13T07:42:09.706331Z
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499)
	at org.projectnessie.gc.base.TestInstantConversion.testInstantConversion(TestInstantConversion.java:37)
```

### PR DESCRIPTION
**Pull request by ajantha-bhat**
Handled the nanosecond precision instant in testcase.

Fixes #3945

### CODE REVIEW

**Review by snazy — CHANGES_REQUESTED** [review]
The test case is legit.
I don't think that removing a failing test case fixes a test bug.

File: gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java [hunk 1]
```diff
@@ -17,23 +17,62 @@
...
-    // Instant.now() gives micro second precision by default.
-    // Because of that the below validation can pass.
-    assertThat(instantFromMicros).isEqualTo(now);
+    // Instant is capable of nanosecond precision.
+    // Hence, convert to microsecond precision before comparing.
+    Instant instantInMicros = now.truncatedTo(ChronoUnit.MICROS);
+    assertThat(instantFromMicros).isEqualTo(instantInMicros);
+  }
+
+  private static class NanoClock extends Clock {
```
Comment (ajantha-bhat) [inline gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java:46]: I added this for java 11 handling the nano second instants.

Comment (snazy) [reply gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java:46]: I don't understand why this is needed for the test.

File: gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java [hunk 2]
```diff
@@ -17,23 +17,62 @@
...
 
 public class TestInstantConversion {
 
-  @Test
-  void testInstantConversion() {
-    Instant now = Instant.now();
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  void testInstantConversion(boolean isNanoPrecision) {
+    Instant now = isNanoPrecision ? Instant.now(new NanoClock(Clock.systemUTC())) : Instant.now();
```
Comment (snazy) [inline gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java:34]: Could use a fixed clock

Comment (snazy) [reply gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java:34]: Or even MutableClock

Comment (ajantha-bhat) [reply gc/gc-base/src/test/java/org/projectnessie/gc/base/TestInstantConversion.java]: used a fixed Instant now.

**Pull request merged**